app/controllers/submissions_controller.rb

```diff
@@ -383,67 +383,57 @@ def update_files
 
       log_messages = []
       begin
+
         if new_files.empty?
           # delete files marked for deletion
           delete_files.each do |filename|
-            txn.remove(File.join(assignment_folder, filename),
-                       file_revisions[filename])
-            log_messages.push("Student '#{current_user.user_name}'" +
-                              " deleted file '#{filename}' for assignment" +
-                              " '#{@assignment.short_identifier}'.")
+            file_path_full = File.join(assignment_folder, filename)
+            file_path_relative = File.join(@path, filename)[1..-1] # remove trailing '/' to match required files
```

Comment:
This seems to happen in a couple of places.

I think a better approach is to call [`Pathname.cleanpath`](https://ruby-doc.org/stdlib-2.5.0/libdoc/pathname/rdoc/Pathname.html#method-i-cleanpath) ~~both here~~ and on the actual user input for the assignment's required files.

(**Updated**) Okay I stand by what I said about the user input, but here I actually prefer using [`Pathname.relative_path_from`](https://ruby-doc.org/stdlib-2.5.0/libdoc/pathname/rdoc/Pathname.html#method-i-relative_path_from). It makes the connection between `file_path_full` and `file_path_relative` more obvious.

Reply:
Related: I was confused at first because I didn't realize `assignment_folder` was a path into a subdirectory, not just `@assignment.repository_folder`. If you could rename that variable that would be great. :)

Reply:
1. With user input you mean the assignment config page, right? I can certainly do that
2. Ok, I'll just have to make things work with Pathnames instead of strings
3. Yes

Reply:
I gave up on the .. matching with a regex.